### DISCUSSION TITLE
NavMenu is not visible after installing Blazorise

### DISCUSSION THREAD

**Discussion by cfeze** [Megabit/Blazorise#1942] [Q&A]
Hi guys,

I created a empty project of Blazor WASM and then I tried install Blazorise but I'm getting this error. NavMenu is disappear

![image](https://user-images.githubusercontent.com/7374023/109548121-05c93200-7aab-11eb-9957-aba69fa57630.png)

If I revert all changes, NaVMenu is back

https://github.com/cfeze/BlazoriseDemo

**Comment by stsrki** [comment]
It's because the default Blazor template comes with predefined custom styles located in `site.css` and that will override some of the Bootstrap classes. And it will affect how the nav menu will look. So it will also clash with some of the styles that come with Blazorise.

Just remove `site.css` and go fully Blazorise. You will remove it sooner or later.

**Reply by cfeze** [reply]
Thanks for your answer.

Do you say BlazoriseDemo\wwwroot\css\app.css ?
If I remove it then "Failed to load resource: the server responded with a status of 404 on app.css" comes up.

**Reply by stsrki** [reply]
Yes, `app.css`. It's either `site.css` or `app.css`, depending on whether it's WebAssembly or a server-side project.

Also, remove `<link href="css/app.css" rel="stylesheet" />` from `index.html` or `_Host.cshtml`.

**Reply by cfeze** [reply]
After that App doesn't has styles. 

I think code below is not working as expected.
```
    <link href="_content/Blazorise/blazorise.css" rel="stylesheet" />
    <link href="_content/Blazorise.Bootstrap/blazorise.bootstrap.css" rel="stylesheet" />
```


![image](https://user-images.githubusercontent.com/7374023/109555990-fd75f480-7ab4-11eb-9cf7-c797bc1cd31e.png)

**Reply by stsrki** [reply]
Well, of course, it will not work by itself. You must add some layout and component :)

Look at the documentation https://blazorise.com/docs/ for each component description. And also look at the demo app for reference https://github.com/stsrki/Blazorise/tree/master/Demos/Blazorise.Demo

**Reply by cfeze** [reply]
Hi, sorry I didn't realize that I must add some layout and component.
I've looked at documentation and source code for this and I'm still stuck on my App, I have added a layout but I'm getting some issues below. 

1) NavMenu and Header are not aligned.
![image](https://user-images.githubusercontent.com/7374023/109681969-c3602d80-7b5c-11eb-8b5d-6e59810887d8.png)

2) An error always is displayed but in Console debugger is OK.
![image](https://user-images.githubusercontent.com/7374023/109682162-ea1e6400-7b5c-11eb-832b-0e8171a550a0.png)

I couldn't resolve them for now. Could you please provide an example in order how to use Blazorise on default project for Blazor WASM?

You can have look at https://github.com/cfeze/BlazoriseDemo if you want.

Thanks!

**Reply by mic-dev-x** [reply]
Hi,
I'va also experieced this strange problem and I've solved it in this way:
in Shared/NavMenu.razor
get rid of the "@onclick="ToggleNavMenu"

**Comment by stsrki** [comment]
1. This is working fine as far as I can see. The **Home** item has slight padding and is a little below the header. That is how it works.
2. Remove this https://github.com/cfeze/BlazoriseDemo/blob/master/BlazoriseDemo/wwwroot/index.html#L20, I personally always delete it.

**Reply by cfeze** [reply]
Great! Thanks. 
Just one thing more. At point 1 How can I fix it? If you see https://github.com/cfeze/BlazoriseDemo/blob/master/BlazoriseDemo/Shared/MainLayout.razor#L20 I just use layout attributes. Do I have to change css files for that?

**Reply by stsrki** [reply]
I really don't know why it won't work for you. As you can see in the demos it's just Blazorise without any custom CSS. You can try changing CSS to adjust the height of the top Bar, or whatever you need. It's the right thing to do.

**Comment by stsrki** [comment]
Wait, I just realized you're talking about the `NavMenu`. The component that comes with a default Blazor project template. Yeah, that will not work because it uses some custom CSS that will clash with Blazorise styles. You will need to go fully Blazorise and use Bar with horizontal mode.
